amqp/readers_writers.go

```diff
@@ -92,19 +92,27 @@ func WriteFrame(wr io.Writer, frame *Frame) (err error) {
 	}
 
 	return nil
-
-	//return binary.Write(wr, binary.BigEndian, frameBuffer.Bytes())
 }
 
 // ReadOctet reads octet (byte)
 func ReadOctet(r io.Reader) (data byte, err error) {
-	err = binary.Read(r, binary.BigEndian, &data)
+	b := make([]byte, 1)
+	if _, err = io.ReadFull(r, b); err != nil {
+		return
+	}
+	data = b[0]
 	return
 }
 
 // WriteOctet writes octet (byte)
 func WriteOctet(wr io.Writer, data byte) error {
-	return binary.Write(wr, binary.BigEndian, data)
+	b := make([]byte, 1)
```

Comment:
Same for the writer, I'd think allocating the slice on the stack will likely improve performance

Reply:
fixed